meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh

```diff
@@ -967,3 +989,122 @@ set_intel_regions_update_params() {
     fi
   fi
 }
+
+handle_fw_switching() {
+  local _can_switch_to_heads=$1
+
+  if [ "$_can_switch_to_heads" == "true" ] && [ "$DASHARO_FLAVOR" != "Dasharo (coreboot+heads)" ]; then
+    while : ; do
+      echo
+      read -r -p "Would you like to switch to Dasharo heads firmware? (Y|n) " OPTION
+      echo
+
+      case ${OPTION} in
+        yes|y|Y|Yes|YES)
+          UPDATE_VERSION=$HEADS_REL_VER_DES
+          FLASHROM_ADD_OPT_UPDATE_OVERRIDE="--ifd -i bios"
+          BIOS_HASH_LINK="${HEADS_LINK_DES}.sha256"
+          BIOS_SIGN_LINK="${HEADS_LINK_DES}.sha256.sig"
+          BIOS_LINK=$HEADS_LINK_DES
+          echo
+          echo "Switching to Dasharo heads firmware v$UPDATE_VERSION"
+          echo
+          break
+          ;;
+        n|N)
+          compare_versions $DASHARO_VERSION $UPDATE_VERSION
+          if [ $? -ne 1 ]; then
+            error_exit "No update available for your machine"
+          fi
+          echo "Will not install Dasharo heads firmware. Proceeding with regular Dasharo firmware update."
+          break
+          ;;
+        *)
+          ;;
+      esac
+    done
+  elif [ -v DES_IS_LOGGED ] && [ -v HEADS_LINK_DES ]; then
+    [ "$DASHARO_FLAVOR" == "Dasharo (coreboot+heads)" ]
```

Comment:
@miczyg1 what is the purpose of this?

Reply:
i have a proper IF condition in line 1032, so I believe I left some copy-pasta accidentally.